cmd/sovereignnode/chainSimulator/tests/vm/issue_test.go

```diff
@@ -0,0 +1,213 @@
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/components/api"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator/dtos"
+	sovereignChainSimulator "github.com/multiversx/mx-chain-go/sovereignnode/chainSimulator"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	coreAPI "github.com/multiversx/mx-chain-core-go/data/api"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	defaultPathToInitialConfig = "../../../../node/config/"
+	sovereignConfigPath        = "../../../config/"
+	issueWasmPath              = "../testdata/issue.wasm"
+	issuePrice                 = "5000000000000000000"
+)
+
+var oneEgld = big.NewInt(1000000000000000000)
+var initialMinting = big.NewInt(0).Mul(oneEgld, big.NewInt(100))
+
+func TestSmartContract_IssueToken(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	cs, err := sovereignChainSimulator.NewSovereignChainSimulator(sovereignChainSimulator.ArgsSovereignChainSimulator{
+		SovereignConfigPath: sovereignConfigPath,
+		ChainSimulatorArgs: &chainSimulator.ArgsChainSimulator{
+			BypassTxSignatureCheck: false,
+			TempDir:                t.TempDir(),
+			PathToInitialConfig:    defaultPathToInitialConfig,
+			NumOfShards:            1,
+			GenesisTimestamp:       time.Now().Unix(),
+			RoundDurationInMillis:  uint64(6000),
+			RoundsPerEpoch:         core.OptionalUint64{},
+			ApiInterface:           api.NewNoApiInterface(),
+			MinNodesPerShard:       2,
+			ConsensusGroupSize:     2,
+			AlterConfigsFunction: func(cfg *config.Configs) {
+				cfg.SystemSCConfig.ESDTSystemSCConfig.BaseIssuingCost = issuePrice
+			},
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+
+	defer cs.Close()
+
+	err = cs.GenerateBlocks(200)
+	require.Nil(t, err)
+
+	nodeHandler := cs.GetNodeHandler(core.SovereignChainShardId)
+	systemScAddress, _ := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Decode("erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu")
+
+	wallet, err := cs.GenerateAndMintWalletAddress(core.SovereignChainShardId, initialMinting)
+	require.Nil(t, err)
+	nonce := uint64(0)
+
+	deployedContractAddress := chainSimulatorIntegrationTests.DeployContract(t, cs, wallet.Bytes, &nonce, systemScAddress, "", issueWasmPath)
+	deployedContractAddressBech32, err := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Encode(deployedContractAddress)
+	require.Nil(t, err)
+
+	issueCost, _ := big.NewInt(0).SetString(issuePrice, 10)
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, deployedContractAddress, issueCost, "issue", uint64(60000000))
+
+	accountSC, _, err := nodeHandler.GetFacadeHandler().GetAccount(deployedContractAddressBech32, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.True(t, accountSC.Balance == "0")
+
+	esdts, err := nodeHandler.GetFacadeHandler().GetAllIssuedESDTs(core.FungibleESDT)
+	require.Nil(t, err)
+	require.NotNil(t, esdts)
+	require.True(t, len(esdts) == 1)
```

Comment:
Let's also check the initial supply

Reply:
Cannot check initial supply because there was no mint done